### PR TITLE
feat: AI - improve writing

### LINKED ISSUE #2039: [FR] AI - Improve writing
### Description

1. select a range of continuous text.
2. click the 💡 icon on the toolbar.
3. select the action `improve writing`.
4. call the OpenAI service to improve the selected text writing.

Screenshot 2023-03-20 at 21 24 28

### Impact

The users who want to improve the selected text writing.

### Additional Context

* refer to `smart_edit_action.dart`
* https://platform.openai.com/examples

### PR DESCRIPTION
**Pull request by alceil**
### Feature Preview
[Untitled_ Apr 13, 2023 5_07 PM.webm](https://user-images.githubusercontent.com/47685349/231747452-b46177ea-e2a2-4192-8d3f-1c18f2b601be.webm)

Fixes #2039 

#### PR Checklist

- [x] My code adheres to the [AppFlowy Style Guide](https://appflowy.gitbook.io/docs/essential-documentation/contribute-to-appflowy/software-contributions/submitting-code/style-guides)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [x] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [x] All existing tests are passing.

### CODE REVIEW

**Comment by LucasXu0** [comment]
LGTM. Cool, @alceil. By the way, can you help to address this UI issue as the image below shows. There's a new line in front of the result.

Screenshot 2023-04-14 at 10 22 51

**Comment by alceil** [comment]
Sure @LucasXu0

**Comment by alceil** [comment]
I have added some relevant changes can you have a look @LucasXu0

**Comment by LucasXu0** [comment]
@alceil. So, is the problem that the result from OpenAI returns '\n' as the first line? And I think you should format the code.

**Comment by alceil** [comment]
Ya @LucasXu0  it returns a newline for code completion api mostly. Should I try fixing it using another approach. Would like your take on it!

**Comment by alceil** [comment]
I have formatted the code is there something else left to do @LucasXu0

File: frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/openai/widgets/smart_edit_action.dart [hunk 1]
```diff
@@ -22,6 +25,8 @@ enum SmartEditAction {
         return '$input\n\nTl;dr';
       case SmartEditAction.fixSpelling:
         return 'Correct this to standard English:\n\n$input';
+      case SmartEditAction.improveWriting:
+        return 'Rewrite this:\n\n$input';
```
Comment (Xazin) [inline frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/openai/widgets/smart_edit_action.dart:29]: Do we not want to make it more implicit that the writing should be improved? Or is that already default behavior.

Eg. `Improve the writing of:`

Comment (alceil) [reply frontend/appflowy_flutter/lib/plugins/document/presentation/plugins/openai/widgets/smart_edit_action.dart:29]: I think giving this prompt gets the desired result

**Pull request merged**